From ed67d6e79cbfc5699301505eb3c5df929aeaf91e Mon Sep 17 00:00:00 2001
From: Oleh Kuzhylnyi <kuzhylol@gmail.com>
Date: Sat, 17 May 2025 20:47:12 +0200
Subject: [PATCH] disable slide show

---
 src/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index 65ca2db..dd8e47e 100644
--- a/src/main.c
+++ b/src/main.c
@@ -152,7 +152,7 @@ int main(int argc, char **argv)
 
     /*Create a Demo*/
     lv_demo_widgets();
-    lv_demo_widgets_start_slideshow();
+    //lv_demo_widgets_start_slideshow();
 
     /* Enter the run loop of the selected backend */
     driver_backends_run_loop();
